src/drivers/fluid_pipewire.c

```diff
@@ -0,0 +1,299 @@
+        return NULL;
+    }
+
+    buffer = FLUID_ARRAY(float, NUM_CHANNELS *period_size);
+    if (!buffer)
+    {
+        FLUID_LOG(FLUID_ERR, "Out of memory");
+        goto driver_cleanup;
+    }
+    buffer_length = period_size * sizeof(float) * NUM_CHANNELS;
+
+    drv->builder = FLUID_NEW(struct spa_pod_builder);
+    if (!drv->builder)
+    {
+        FLUID_LOG(FLUID_ERR, "Out of memory");
+        goto driver_cleanup;
+    }
+    FLUID_MEMSET(drv->builder, 0, sizeof(*drv->builder));
+    drv->builder->data = buffer;
+    drv->builder->size = buffer_length;
+
+    if (func)
+    {
+        drv->lbuf = FLUID_ARRAY(float, period_size);
+        drv->rbuf = FLUID_ARRAY(float, period_size);
+
+        if (!drv->lbuf || !drv->rbuf)
+        {
+            FLUID_LOG(FLUID_ERR, "Out of memory");
+            goto driver_cleanup;
+        }
+    }
+
+    params[0] = spa_format_audio_raw_build(drv->builder,
+                                           SPA_PARAM_EnumFormat,
+                                           &SPA_AUDIO_INFO_RAW_INIT(.format = SPA_AUDIO_FORMAT_F32,
+                                                                    .channels = 2,
+                                                                    .rate = sample_rate));
+
+    pw_flags = PW_STREAM_FLAG_AUTOCONNECT | PW_STREAM_FLAG_MAP_BUFFERS;
+    pw_flags |= realtime_prio ? PW_STREAM_FLAG_RT_PROCESS : 0;
+    res = pw_stream_connect(drv->pw_stream, PW_DIRECTION_OUTPUT, PW_ID_ANY, pw_flags, params, 1);
+    if (res < 0)
+    {
+        FLUID_LOG(FLUID_ERR, "PipeWire stream connection failed");
+        goto driver_cleanup;
+    }
+
+    res = pw_thread_loop_start(drv->pw_loop);
+    if (res != 0)
+    {
+        FLUID_LOG(FLUID_ERR, "Failed starting PipeWire loop");
+        goto driver_cleanup;
+    }
+
+    FLUID_LOG(FLUID_INFO, "Using PipeWire audio driver");
+
+    return (fluid_audio_driver_t *)drv;
+
+driver_cleanup:
+    delete_fluid_pipewire_audio_driver((fluid_audio_driver_t *)drv);
+    return NULL;
+}
+
+void delete_fluid_pipewire_audio_driver(fluid_audio_driver_t *p)
+{
+    fluid_pipewire_audio_driver_t *drv = (fluid_pipewire_audio_driver_t *)p;
+    fluid_return_if_fail(drv);
+
+    if (drv->pw_stream)
+    {
+        pw_stream_destroy(drv->pw_stream);
+    }
+
+    if (drv->pw_loop)
+    {
+        pw_thread_loop_destroy(drv->pw_loop);
+    }
+
+    FLUID_FREE(drv->lbuf);
+    FLUID_FREE(drv->rbuf);
+
+    if (drv->builder)
+    {
+        FLUID_FREE(drv->builder->data);
+    }
+    FLUID_FREE(drv->builder);
+
+    FLUID_FREE(drv->events);
+
+    FLUID_FREE(drv);
+}
+
+void fluid_pipewire_audio_driver_settings(fluid_settings_t *settings)
+{
+    fluid_settings_register_str(settings, "audio.pipewire.media-role", "Music", 0);
+    fluid_settings_register_str(settings, "audio.pipewire.media-type", "Audio", 0);
+    fluid_settings_register_str(settings, "audio.pipewire.media-category", "Playback", 0);
```

Comment:
Those new `audio.pipewire.*`  have no options registered (what we usually do with `fluid_settings_add_option()`). Is that intended? If they only ever default to the standard option, do we actually need them?
In case we do need them, they should be documented in /doc/fluidsettings.xml. Just placing a link to the pipewire documentation should be sufficient.

Reply:
Yeah, I didn't set options because application developers can use different values here. e.g. when using fluidsynth as the audio engine in a game, one can set "audio.pipewire.media-role" to "Game" as per [the freedesktop specification](https://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/Developer/Clients/ApplicationProperties/). This is for integration with the system's audio mixer. Not sure what to do here, besides copying all the values from the spec..

Reply:
NP, I'll take care of the documentation after merge.